pkg/storage/types.go

```diff
@@ -24,13 +35,23 @@ func (r *Response) Status() int {
 	return r.status
 }
 
+// Metadata returns metadata field from the struct
+func (r *Response) Metadata() *ResponseMetadata {
+	return r.metadata
+}
+
 // NewResponse takes data, statusCode and error as arguments and returns a new Response
-func NewResponse(data []byte, statusCode int, err error) *Response {
-	return &Response{data: data, err: err, status: statusCode}
+func NewResponse(data []byte, statusCode int, err error, metadata *ResponseMetadata) *Response {
```

Comment:
We should not be changing this API. Since `metadata` is optional, it should be set via separate call.

```
res := NewResponse()
res = SetMetadata(res, metadata)
```

This is similar to how you set values in the context api

Reply:
Or follow  the options pattern `WithMetadata` which adds metadata to existing object